src/filebrowser/tasks.cpp

```diff
@@ -272,6 +272,11 @@ void FileUploadDirectoryTask::createFileServerTask(const QString& link)
     if (local_path_ == "/")
         qWarning("attempt to upload the root directory, you should avoid it\n");
     QDir dir(local_path_);
+    if (dir.count() == 2) {
+        // only contains . and .., so an empty folder
+        empty_folder_ = dir.dirName();
+    }
```

Comment:
I don't think we need this new `empty_folder_` variable. If we find the top level folder is empty, we can simply include it in the `empty_subfolders_` list, so the other part of the code don't need to be changed.